src/net.cpp

```diff
@@ -912,6 +912,427 @@ size_t V1Transport::GetSendMemoryUsage() const noexcept
     return m_message_to_send.GetMemoryUsage();
 }
 
+V2Transport::V2Transport(NodeId nodeid, bool initiating, int type_in, int version_in) noexcept :
+    m_cipher{}, m_initiating{initiating}, m_nodeid{nodeid},
+    m_recv_type{type_in}, m_recv_version{version_in},
+    m_recv_state{RecvState::KEY},
+    m_send_state{SendState::AWAITING_KEY}
+{
+    // Initialize the send buffer with ellswift pubkey.
+    m_send_buffer.resize(EllSwiftPubKey::size());
+    std::copy(std::begin(m_cipher.GetOurPubKey()), std::end(m_cipher.GetOurPubKey()), MakeWritableByteSpan(m_send_buffer).begin());
+}
+
+V2Transport::V2Transport(NodeId nodeid, bool initiating, int type_in, int version_in, const CKey& key, Span<const std::byte> ent32) noexcept :
+    m_cipher{key, ent32}, m_initiating{initiating}, m_nodeid{nodeid},
+    m_recv_type{type_in}, m_recv_version{version_in},
+    m_recv_state{RecvState::KEY},
+    m_send_state{SendState::AWAITING_KEY}
```

Comment:
follow-up consideration: the two V2Transport constructors have a lot of code in common, is it somehow possible to deduplicate their implementation (e.g. with another private constructor)?

Reply:
I played with this some time ago. I used a private constructor that is called by the other two public constructors (see [delegating constructor](https://stroustrup.com/C++11FAQ.html#delegating-ctor)). One problem was that `BIP324Cipher` is not movable, so I involved `std::forward` to pass the arguments from the public ctors to the private one which would construct the `BIP324Cipher` object. It worked. Also the private ctor had to have different arguments than the public ones, so I added a dummy `nullptr_t` first arg (:disappointed:). Then in a later commit of this PR the two public ctors diverged and were not identical anymore, so I decided to ditch the idea.